### PR TITLE
Use IndexMap instead of LinkedHashMap

### PR DESCRIPTION
**Pull request by lnicola**
This was surprisingly easy :confused:.

### CODE REVIEW

**Comment by lnicola** [comment]
Is there a benchmark suite or something like that? I wonder if we could drop the other implementation.

**Comment by lnicola** [comment]
Oh, `indexmap` requires 1.18.

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ appveyor = { repository = "serde-rs/json" }
 
 [dependencies]
 serde = "1.0.60"
-linked-hash-map = { version = "0.5", optional = true }
+indexmap = { version = "1.0", optional = true }
```
Comment (dtolnay) [inline Cargo.toml]: Please change the Travis configuration to add a build at indexmap's minimum supported compiler version, and remove any `preserve_order` builds below that compiler version.

Comment (lnicola) [reply Cargo.toml]: Done

**Pull request merged**